app/frontend/actions/wallet.ts

```diff
@@ -170,12 +172,14 @@ export default (store: Store) => {
     loadingAction(state, 'Reloading wallet info...')
     try {
       const accountsInfo = await wallet.getAccountsInfo(state.validStakepoolDataProvider)
+      const tokensMetadata = await wallet.getTokensMetadata(accountsInfo)
```

Comment:
what about tokens present in transaction history that are no longer part of the current balance of the account?

Reply:
Fixed, transaction history is now considered as well.